website/source/docs/secrets/ad/index.html.md

```diff
@@ -0,0 +1,172 @@
+twice as often as its TTL.
+
+## A Note on Escaping
+
+**It is up to the administrator** to provide properly escaped DNs. This
+includes the user DN, bind DN for search, and so on.
+
+The only DN escaping performed by this method is on usernames given at login
+time when they are inserted into the final bind DN, and uses escaping rules
+defined in RFC 4514.
+
+Additionally, Active Directory has escaping rules that differ slightly from the
+RFC; in particular it requires escaping of '#' regardless of position in the DN
+(the RFC only requires it to be escaped when it is the first character), and
+'=', which the RFC indicates can be escaped with a backslash, but does not
+contain in its set of required escapes. If you are using Active Directory and
+these appear in your usernames, please ensure that they are escaped, in
+addition to being properly escaped in your configured DNs.
+
+For reference, see [RFC 4514](https://www.ietf.org/rfc/rfc4514.txt) and this
+[TechNet post on characters to escape in Active
+Directory](http://social.technet.microsoft.com/wiki/contents/articles/5312.active-directory-characters-to-escape.aspx).
+
+## Quick Setup
+
+Most secrets engines must be configured in advance before they can perform their
+functions. These steps are usually completed by an operator or configuration
+management tool.
+    
+1. Enable the Active Directory secrets engine:
+
+    ```text
+    $ vault secrets enable ad
+    Success! Enabled the ad secrets engine at: ad/
+    ```
+
+    By default, the secrets engine will mount at the name of the engine. To
+    enable the secrets engine at a different path, use the `-path` argument.
+
+2. Configure the credentials that Vault uses to communicate with Active Directory 
+to generate passwords:
+
+    ```text
+    $ vault write ad/config \
+        binddn=$USERNAME \
+        bindpass=$PASSWORD \
+        url=ldap://138.91.247.105 \
+        userdn='dc=example,dc=com'
+    ```
+
+    The `$USERNAME` and `$PASSWORD` given must be of a high enough access level that
+    they can be used for modifying passwords. Typically, this will be a domain admin.
+    
+    If you'd like to do a quick, insecure evaluation, also set `insecure_tls` to true. However, this is NOT RECOMMENDED
+    in a production environment. In production, we recommend `insecure_tls` is false (its default) and is used with a valid 
+    `certificate`.
+
+3. Configure a role that maps a name in Vault to an account in Active Directory.
+When applications request passwords, password rotation settings will be managed by
+this role.
+
+    ```text
+    $ vault write ad/roles/my-application \
+        service_account_name="my-application@example.com"
+    ```
+    
+4. Grant "my-application" access to its creds at `ad/creds/my-application` using an 
+auth method like [AppRole](https://www.vaultproject.io/api/auth/approle/index.html).
+
+## Configuration
```

Comment:
Leave this in the API section, don't duplicate it all here

Reply:
KK.